refactor(linter): extract nested config searching to a fn

camc314 opened the pull request:
moves the nested config code out into a functin to improve readability + prep for future changes

camc314 commented:
* **#10861** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10861?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#10860** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10860?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10860?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10860).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on apps/oxlint/src/lint.rs:
```diff
@@ -166,71 +169,16 @@ impl Runner for LintRunner {
-        let mut nested_oxlintrc = FxHashMap::<&Path, Oxlintrc>::default();
-        let mut nested_configs = FxHashMap::<PathBuf, ConfigStore>::default();
-
         let handler = GraphicalReportHandler::new();
```
NIT: 
The handler is only used for the get_nested_configs. 
Other functions are creating their own handler too.

camc314 replied:
will follow up with this later.

i'm going to make the caller report the diagnostic, not inside the fn

camc314 committed:
refactor(linter): extract nested config searching to a fn (#10860)

moves the nested config code out into a functin to improve readability + prep for future changes
